plugin/pkg/scheduler/core/scheduling_queue_test.go

```diff
@@ -0,0 +1,303 @@
+			Conditions: []v1.PodCondition{
+				{
+					Type:   v1.PodScheduled,
+					Status: v1.ConditionFalse,
+					Reason: v1.PodReasonUnschedulable,
+				},
+			},
+		},
+	}
+
+func TestPriorityQueue_Add(t *testing.T) {
+	q := NewPriorityQueue()
+	q.Add(&medPriorityPod)
+	q.Add(&unschedulablePod)
+	q.Add(&highPriorityPod)
+	if p, err := q.Pop(); err != nil || p != &highPriorityPod {
+		t.Errorf("Expected: %v after Pop, but got: %v", highPriorityPod.Name, p.Name)
+	}
+	if p, err := q.Pop(); err != nil || p != &medPriorityPod {
+		t.Errorf("Expected: %v after Pop, but got: %v", medPriorityPod.Name, p.Name)
+	}
+	if p, err := q.Pop(); err != nil || p != &unschedulablePod {
+		t.Errorf("Expected: %v after Pop, but got: %v", unschedulablePod.Name, p.Name)
+	}
+}
+
+func TestPriorityQueue_Pop(t *testing.T) {
+	q := NewPriorityQueue()
+	go func() {
+		if p, err := q.Pop(); err != nil || p != &highPriorityPod {
+			t.Errorf("Expected: %v after Pop, but got: %v", highPriorityPod.Name, p.Name)
+		}
+	}()
+	q.Add(&highPriorityPod)
+}
+
+func TestPriorityQueue_Update(t *testing.T) {
+	q := NewPriorityQueue()
+	q.Update(&highPriorityPod)
+	if _, exists, _ := q.activeQ.Get(&highPriorityPod); !exists {
+		t.Errorf("Expected %v to be added to activeQ.", highPriorityPod.Name)
+	}
+	q.Update(&highPriorityPod)
+	if q.activeQ.data.Len() != 1 {
+		t.Error("Expected only one item in activeQ.")
+	}
+	// Updating an unschedulable pod which is not in any of the two queues, should
+	// add the pod to unschedulableQ.
+	q.Update(&unschedulablePod)
+	if len(q.unschedulableQ.pods["node1"]) != 1 {
+		t.Errorf("Expected %v to be added to unschedulableQ.", unschedulablePod.Name)
+	}
+	// Updating a pod that is already in unschedulableQ, should move the pod to
+	// activeQ.
+	q.Update(&unschedulablePod)
+	if len(q.unschedulableQ.pods) != 0 {
+		t.Error("Expected unschedulableQ to be empty.")
+	}
+	if _, exists, _ := q.activeQ.Get(&unschedulablePod); !exists {
+		t.Errorf("Expected: %v to be added to activeQ.", unschedulablePod.Name)
+	}
+	if p, err := q.Pop(); err != nil || p != &highPriorityPod {
+		t.Errorf("Expected: %v after Pop, but got: %v", highPriorityPod.Name, p.Name)
+	}
+}
+
+func TestPriorityQueue_Delete(t *testing.T) {
+	q := NewPriorityQueue()
+	q.Update(&highPriorityPod)
+	q.Add(&unschedulablePod)
+	q.Delete(&highPriorityPod)
+	if _, exists, _ := q.activeQ.Get(&unschedulablePod); !exists {
+		t.Errorf("Expected %v to be in activeQ.", unschedulablePod.Name)
+	}
+	if _, exists, _ := q.activeQ.Get(&highPriorityPod); exists {
+		t.Errorf("Didn't expect %v to be in activeQ.", highPriorityPod.Name)
+	}
+}
+
+func TestPriorityQueue_MoveAllToActiveQueue(t *testing.T) {
+	q := NewPriorityQueue()
+	q.Add(&medPriorityPod)
+	q.unschedulableQ.Add(&unschedulablePod)
+	q.unschedulableQ.Add(&highPriorityPod)
+	q.MoveAllToActiveQueue()
+	if q.activeQ.data.Len() != 3 {
+		t.Error("Expected all items to be in activeQ.")
+	}
+}
+
+func TestPriorityQueue_AssignedPodAdded(t *testing.T) {
```

Comment:
I'm a fan of describing the conditions being tested with comments, as I have to task switch often and it takes a while to unpack some of this.

Reply:
The test is testing PriorityQueue.AssignedPodAdded(). Are you suggesting to add a test scenario?

Reply:
I added more comments.